faiss/python/class_wrappers.py

```diff
@@ -226,7 +226,10 @@ def replacement_add(self, x):
 
         n, d = x.shape
         assert d == self.d
-        x = np.ascontiguousarray(x, dtype='float32')
+        if numeric_type == faiss.Float32:
+            x = np.ascontiguousarray(x, dtype='float32')
+        else:
+            x = np.ascontiguousarray(x, dtype='float16')
         self.add_c(n, swig_ptr(x))
```

Comment:
this will not work in the fp16 case because it will go through the regular add method, not the one with numeric_type